### ISSUE TITLE
Cannot add image to Grid in 8.12

### ISSUE THREAD

**Issue by christophborndigital** [umbraco/Umbraco-CMS#9982]
After upgrading a site from 8.11.1 to 8.12, I can no longer add an image to a Grid editor using the standard "Image" grid control.

## Umbraco version

- 8.12.0
- 8.12.1

Reproduction
------------

### Specifics

After selecting an image, another infinity editor opens "Edit selected media".
Clicking Submit causes a JS error in `umbraco.controllers.js` `function updateControlValue(selectedImage)` on line 18732 in 8.12.1:

Error: "$scope.control.value is null"

### Steps to reproduce

- Edit a content item with a Grid editor
- Add an `Image` control
- Click "Click to insert image"
- Select an image
- Another infinity editor pops up "Edit selected media"
- Click "Select"
- Nothing happens (apart from a console JS error)

### Expected result

Image added to the grid

### Actual result

Image isn't selected, I can only "Close" the dialog

**Comment by Jay-umbr** [comment]
Can reproduce, seems to be an issue not only on upgraded projects but also on new ones.

**Comment by ChristophClaus** [comment]
Can reproduce! Existential error...

**Comment by nul800sebastiaan** [comment]
I've just noticed this as well 😱 We'll look into it!

**Comment by nul800sebastiaan** [comment]
As a workaround for now, you can update the `updateControlValue` method in `Umbraco\js\umbraco.controllers.js` to:

```javascript
        function updateControlValue(selectedImage) {
            // const doGetThumbnail = $scope.control.value.focalPoint !== selectedImage.focalPoint
            //     || $scope.control.value.image !== selectedImage.image;

            // we could apply selectedImage directly to $scope.control.value,
            // but this allows excluding fields in future if needed
            $scope.control.value = {
                focalPoint: selectedImage.focalPoint,
                coordinates: selectedImage.coordinates,
                id: selectedImage.id,
                udi: selectedImage.udi,
                image: selectedImage.image,
                caption: selectedImage.caption,
                altText: selectedImage.altText
            };


            // if (doGetThumbnail) {
                $scope.thumbnailUrl = getThumbnailUrl();
            //}
        }
```

This method was added in @nathanwoulfe 's PR https://github.com/umbraco/Umbraco-CMS/pull/9681/files#diff-16bf5e50f31cc4c3ef3f2af6bf0a399b98e5e2e926e793184cffdc9d0f0ecbcdR105

I don't know how this was supposed to work, since `$scope.control.value` is null to begin with, so it should have errored out.

**Comment by nathanwoulfe** [comment]
@nul800sebastiaan  PR to fix is in #9987 - shouldn't have been attempting to access control values until they'd been set.

**Comment by nul800sebastiaan** [comment]
That was fast @nathanwoulfe! Thanks! Will have a look tomorrow!

**Comment by nul800sebastiaan** [comment]
Fixed in https://github.com/umbraco/Umbraco-CMS/pull/9987 for the upcoming 8.12.2.

**Comment by christophborndigital** [comment]
Thanks for the prompt fix, looking forward to 8.12.2
